Tools/cases_generator/analyzer.py

```diff
@@ -320,10 +340,17 @@ def tier_variable(node: parser.InstDef) -> int | None:
                 return int(token.text[-1])
     return None
 
-def is_infallible(op: parser.InstDef) -> bool:
-    return not (
+def has_pop_error(op: parser.InstDef) -> bool:
+    return (
         variable_used(op, "ERROR_IF")
-        or variable_used(op, "error")
+        or variable_used(op, "pop_1_error")
+        or variable_used(op, "exception_unwind")
+        or variable_used(op, "resume_with_error")
+    )
+
+def has_no_pop_error(op: parser.InstDef) -> bool:
+    return (
+        variable_used(op, "NO_POP_ERROR")
         or variable_used(op, "pop_1_error")
         or variable_used(op, "exception_unwind")
         or variable_used(op, "resume_with_error")
```

Comment:
Hm. Shouldn't these other three be only in the `has_pop_error` check?

Reply:
It isn't clear. If there are explicit gotos in the C code, it is hard to tell. It is safer to assume that it could be either.